src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] record = line.split(",");
+                String operationType = record[0];
+                int amount = Integer.parseInt(record[1]);
+                if ("supply".equals(operationType)) {
+                    totalSupply += amount;
+                } else if ("buy".equals(operationType)) {
+                    totalBuy += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Specified exception text: " + System.lineSeparator(), e);
+        }
+
+        int result = totalSupply - totalBuy;
+        StringBuilder stats = new StringBuilder()
+                .append("supply,").append(totalSupply).append(System.lineSeparator())
+                .append("buy,").append(totalBuy).append(System.lineSeparator())
+                .append("result,").append(result);
 
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName,false))) {
+            writer.write(stats.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Specified exception text: " + System.lineSeparator(), e);
+        }
     }
```

Comment:
The `getStatistic` method is doing too much by reading from a file, processing data, and then writing to another file. It would be better to divide this logic into separate methods to improve readability and maintainability. For example, you could have one method to read and process the file and another to write the statistics to the output file.